### DISCUSSION TITLE
How to properly use/mount collections?

### DISCUSSION THREAD

**Discussion by apunkt** [seaweedfs/seaweedfs#5779] [Q&A] [answered]
My evaluation setup:

`./weed master -volumeSizeLimitMB=102`
`./weed volume -max=0 -mserver=192.168.0.206:9333 -dir=./brick -ip=192.168.0.206 -port=8090`
`./weed filer`

Now I add collections
`curl http://192.168.0.206:9333/dir/assign?collection=col0`
`curl http://192.168.0.206:9333/dir/assign?collection=col1`

now mount col0:
`./weed mount -filer=192.168.0.206:8888 -dir=/mnt/tmp -collection=col0`

put files into /mnt/tmp

now mount col1:
`./weed mount -filer=192.168.0.206:8888 -dir=/mnt/tmp2 -collection=col1`

Expectation: /mnt/tmp2 is empty
but is not! Showing files from collection=col0 instead!

Any fops on /mnt/tmp2 now triggers new set of volume creation with empty name! Why?
These new volumes contain files! Which?
New volumes (that yet don't contain files) cannot be removed with volume.deleteEmpty, only with volume.delete and specific id.

What am I doing wrong?
Do I need to mount collections with filer.path?
Also tried to put collections in different dirs then mount with specified dir in filer.path argument. Even a file read then triggers volume creation with empty collection name...

**Comment by chrislusf** [comment]
collection is to create new files.

```
weed mount -h
Example: weed mount -filer=localhost:8888 -dir=/some/dir
Default Usage:
  ...
  -collection string
    	collection to create the files
```

**Reply by apunkt** [reply]
This part is working @chrislusf !
When I have the above two mount points mounted with two different collections, it works, but at the same time a new set of volumes is created with no collection label (empty) that holds apparently the same amount of files?!
So I have volumes for collection 'col0', 'col1' and '' (empty). 3 different volumes, where expectation is 2.

When deleting the empty labled ones, they reappear even when just reading a file from collection=col2 mount point!
Why? I clearly must do something wrong here, I cannot explain the behaviour of new volume creation when mounting two collections and just reading a file from it. Volume creation also must not be triggered as all volumes are basically holding just a few MBs of files.

**Reply by chrislusf** [reply, marked as answer]
there is a default collection for metadata logs.
